nimble does not compile on nrf52

<!--
====================================
IF YOUR ISSUE IS RELATED TO SECURITY
====================================
please submit it to the security mailing-list security@riot-os.org.

If your issue is a question related to the usage of RIOT, please submit it to
our forum at https://forum.riot-os.org.
-->

#### Description
<!--
Example: Cannot build gnrc_networking application for samr21-xpro board.
-->

On the current master branch, nimble fails to compile for the `nrf52840dk` board.
The same seems to apply for any nrf52, the `nrf52dk`, which is the default in the example also suffers from this.

Problem seems to be that there are two nrfx packages being built; the pkg_nrfx, but [RIOT]/build/nrf5x_nrfx_mdk also contains a nrfx header. The latter is a newer version which contains the problematic haly folder, while the actual package does not.

It worked around 1 week ago, altough I cannot say how outdated my local repo was at that point.

#### Steps to reproduce the issue
<!--
Try to describe as precisely as possible here the steps required to reproduce
the issue. Here you can also describe your hardware configuration, the network
setup, etc.
-->

Try to compile the `examples/networking/ble/nimble/nimble_gatt` example.

#### Expected results
<!--
Example: The gnrc_networking application builds on samr21-xpro.
-->
The build succeeds.

#### Actual results
<!--
Please paste or specifically describe the actual output.
-->

```
...
"make" -C /home/simon/Code/dtn/ref-repos/RIOT/build/pkg/nimble/nimble/drivers/nrf52/src/ -f /home/simon/Code/dtn/ref-repos/RIOT/Makefile.base MODULE=nimble_drivers_nrf5x
In file included from /home/simon/Code/dtn/ref-repos/RIOT/build/pkg/nimble/nimble/drivers/nrf52/src/ble_hw.c:28:
/home/simon/Code/dtn/ref-repos/RIOT/build/nrf5x_nrfx_mdk/nrf52/nrfx.h:41:10: fatal error: haly/nrfy_common.h: No such file or directory
   41 | #include <haly/nrfy_common.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

#### Versions
<!--
Operating system: macOS, Linux, Vagrant VM
Build environment: GCC, CLang versions (you can run the following command from
the RIOT base directory: make print-versions).
-->

<details>
<summary>Click to expand</summary>

```
RIOT version information
----------------------------
              RIOT branch: master
         RIOT commit hash: 78468529b4433ca70a6b65ed15d170f50b0918ea
         RIOT commit date: 2026-02-26

Operating System Environment
----------------------------
         Operating System: "Ubuntu" "24.04.4 LTS (Noble Numbat)"
                   Kernel: Linux 6.17.0-1011-oem x86_64 x86_64
             System shell: /usr/bin/dash (probably dash)
             make's shell: /usr/bin/dash (probably dash)

Installed compiler toolchains
-----------------------------
               native gcc: gcc (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
        arm-none-eabi-gcc: arm-none-eabi-gcc (15:13.2.rel1-2) 13.2.1 20231009
                  avr-gcc: missing
           msp430-elf-gcc: missing
       riscv-none-elf-gcc: missing
  riscv64-unknown-elf-gcc: missing
      riscv32-esp-elf-gcc: missing
  riscv32-unknown-elf-gcc: missing
     xtensa-esp32-elf-gcc: missing
   xtensa-esp32s2-elf-gcc: missing
   xtensa-esp32s3-elf-gcc: missing
   xtensa-esp8266-elf-gcc: missing
                    clang: missing

Installed compiler libs
-----------------------
     arm-none-eabi-newlib: "4.4.0"
        msp430-elf-newlib: missing
    riscv-none-elf-newlib: missing
riscv64-unknown-elf-newlib: missing
   riscv32-esp-elf-newlib: missing
riscv32-unknown-elf-newlib: missing
  xtensa-esp32-elf-newlib: missing
xtensa-esp32s2-elf-newlib: missing
xtensa-esp32s3-elf-newlib: missing
xtensa-esp8266-elf-newlib: missing
                 avr-libc: missing (missing)

Installed development tools
---------------------------
                   ccache: missing
                    cmake: cmake version 3.28.3
                 cppcheck: missing
                  doxygen: 1.9.8
                      git: git version 2.43.0
                     make: GNU Make 4.3
                  openocd: Open On-Chip Debugger 0.12.0
                   python: missing
                  python2: missing
                  python3: Python 3.12.3
                   rustup: rustup 1.26.0 (2024-04-01)
                    cargo: cargo 1.91.0 (ea2d97820 2025-10-10)
                    rustc: rustc 1.91.0 (f8297e351 2025-10-28)
                   c2rust: missing
                     node: missing
                      npm: missing
                   flake8: 7.0.0 (mccabe: 0.7.0, pycodestyle: 2.11.1, pyflakes: 3.2.0) CPython 3.12.3 on
               coccinelle: spatch version 1.1.1 compiled with OCaml version 4.14.1
```
</details>

--- a/pkg/pkg.mk
+++ b/pkg/pkg.mk
@@ -171,7 +171,7 @@ $(PKG_SOURCE_DIR)/.git: $(PKG_SPARSE_TAG) | $(PKG_CUSTOM_PREPARED)
 	$(Q)$(GIT_CACHE_RS) clone --commit $(PKG_VERSION) $(addprefix --sparse-add ,$(PKG_SPARSE_PATHS)) -- $(PKG_URL) $(PKG_SOURCE_DIR)
 else ifeq ($(GIT_CACHE_DIR),$(wildcard $(GIT_CACHE_DIR)))
 $(PKG_SOURCE_DIR)/.git: | $(PKG_CUSTOM_PREPARED)
-	$(if $(QUIETER),,$(info [INFO] cloning $(PKG_NAME) with git-cache with git-cache))
+	$(if $(QUIETER),,$(info [INFO] cloning $(PKG_NAME) with git-cache))
 	@echo "$(COLOR_YELLOW)Warning: git-cache is deprecated and will be" \
 	  "removed after the 2026.04 release! See" \
 	  "https://guides.riot-os.org/build-system/advanced_build_system_tricks/#speed-up-builds-with-git-cache-rs" \
@@ -193,11 +193,11 @@ $(PKG_SOURCE_DIR)/.git: $(PKG_SPARSE_TAG) | $(PKG_CUSTOM_PREPARED)
 	$(Q)if [ -n "$(PKG_SPARSE_PATHS)" ]; then \
 	  # iff the package uses sparse paths, initialize the repository as sparse \
 	  $(if $(QUIETER),,echo "[INFO] using sparse checkout";) \
-	  $(GIT_IN_PKG) sparse-checkout set; \
-	  $(GIT_IN_PKG) sparse-checkout add $(PKG_SPARSE_PATHS); \
+	  $(GIT_IN_PKG) sparse-checkout set --no-cone -- $(foreach p,$(PKG_SPARSE_PATHS),"$(p)"); \
 	fi
 	$(Q)$(GIT_IN_PKG) remote add origin $(PKG_URL)
-	$(Q)$(GIT_IN_PKG) config extensions.partialClone origin
+	$(Q)$(GIT_IN_PKG) config remote.origin.promisor true
+	$(Q)$(GIT_IN_PKG) config remote.origin.partialclonefilter blob:none
 	$(Q)$(GIT_IN_PKG) config advice.detachedHead false
 	$(Q)$(GIT_IN_PKG) fetch $(GIT_QUIET) --depth=1 -t --filter=blob:none origin $(PKG_VERSION)
 	$(Q)$(GIT_IN_PKG) checkout $(GIT_QUIET) $(PKG_VERSION) 2>&1 | cat

